pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,162 @@
+	if CSIParam.ForceCephFSKernelClient == "false" {
+		cephfsClientType = csiopv1a1.FuseCephFsClient
+	}
+
+	var periodicity csiopv1a1.PeriodicityType
+	switch cluster.Spec.LogCollector.Periodicity {
+	case "1h", "hourly":
+		periodicity = csiopv1a1.HourlyPeriod
+	case "weekly":
+		periodicity = csiopv1a1.WeeklyPeriod
+	case "monthly":
+		periodicity = csiopv1a1.MonthlyPeriod
+	default:
+		periodicity = csiopv1a1.DailyPeriod
+	}
+
+	opConfig.Spec = csiopv1a1.OperatorConfigSpec{
+		LogLevel: int(CSIParam.LogLevel),
+		DriverSpecDefaults: &csiopv1a1.DriverSpec{
+			Log: &csiopv1a1.LogSpec{
+				LogLevel: int(CSIParam.LogLevel),
+				Rotation: &csiopv1a1.LogRotationSpec{
+					MaxLogSize:  maxLogSize,
+					Periodicity: periodicity,
+				},
+			},
+			//  ImageSet: ,
+			ClusterName:    &cluster.Name,
+			EnableMetadata: &CSIParam.CSIEnableMetadata,
+			GRpcTimeout:    int(CSIParam.GRPCTimeout),
+			// SnapshotPolicy: ,
+			GenerateOMapInfo: &CSIParam.EnableOMAPGenerator,
+			FsGroupPolicy:    k8scsiv1.FileFSGroupPolicy,
+			// Encryption: ,
+			NodePlugin: &csiopv1a1.NodePluginSpec{
+				PodCommonSpec: csiopv1a1.PodCommonSpec{
+					// PrioritylClassName: ,
+					// Labels: ,
+					// Annotations: ,
+					Affinity: &v1.Affinity{
+						NodeAffinity: getNodeAffinity(r.opConfig.Parameters, pluginNodeAffinityEnv, &corev1.NodeAffinity{}),
+					},
+					Tolerations: getToleration(r.opConfig.Parameters, pluginTolerationsEnv, []corev1.Toleration{}),
+				},
+				//UpdateStrategy: ,
+				Resources: csiopv1a1.NodePluginResourcesSpec{
+					// Registrar: ,
+					// Liveness: ,
+					// Addons: ,
+					// Plugin: ,
+				},
+				KubeletDirPath:         CSIParam.KubeletDirPath,
+				EnableSeLinuxHostMount: &CSIParam.EnablePluginSelinuxHostMount,
+			},
+			ControllerPlugin: &csiopv1a1.ControllerPluginSpec{
+				PodCommonSpec: csiopv1a1.PodCommonSpec{
+					// PrioritylClassName: ,
+					// Labels: ,
+					// Annotations: ,
+					Affinity: &v1.Affinity{
+						NodeAffinity: getNodeAffinity(r.opConfig.Parameters, provisionerNodeAffinityEnv, &corev1.NodeAffinity{}),
+					},
+					Tolerations: getToleration(r.opConfig.Parameters, provisionerTolerationsEnv, []corev1.Toleration{}),
+				},
+				Replicas:  &CSIParam.ProvisionerReplicas,
+				Resources: csiopv1a1.ControllerPluginResourcesSpec{
+					// Attacher:,
+					// Snapshotter:,
+					// Resizer:,
+					// Provisioner:,
+					// OMapGenerator:,
+					// Liveness:,
+					// Addons:,
+					// Plugin:,
+				},
+			},
+			// AttachRequired: ,
+			// Liveness: &csiopv1a1.LivenessSpec{},
+			LeaderElection: &csiopv1a1.LeaderElectionSpec{
+				LeaseDuration: int(CSIParam.LeaderElectionLeaseDuration),
+				RenewDeadline: int(CSIParam.LeaderElectionRenewDeadline),
+				RetryPeriod:   int(CSIParam.LeaderElectionRetryPeriod),
+			},
+			DeployCsiAddons:  &CSIParam.EnableCSIAddonsSideCar,
+			CephFsClientType: cephfsClientType,
+			KernelMountOptions: map[string]string{
+				"ms_mode": "secure",
+			},
```

Comment:
This should not be set by default here. we will set this in cephConnection CR

Reply:
removed